pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -785,6 +785,15 @@ func doActivate(ctx *zedmanagerContext, uuidStr string,
 		log.Infof("Waiting for AppNetworkStatus for %s\n", uuidStr)
 		return changed
 	}
+	if ns.Error != "" {
```

Comment:
You could avoid moving this code if zedrouter makes sure to clear PendingCreate/PendingModify when it sees an error. 
That would make more sense.

Reply:
yes, the PendingCreate/Modify are getting cleared on Create/Modify path.

Reply:
will change accordingly, however, error check should be before the ns.Activated check.

Reply:
Agreed